### PR TITLE
Document delete

### LINKED ISSUE #75: REST API
- [ ] Upload documents (and delete)
- [ ] Create link (and delete)
- [ ] Send link
- [ ] Retrieve views

Suggestions for additional functionality that an API should perform are welcomed

### PR DESCRIPTION
**Pull request by Aashish-Upadhyay-101**
Fix #75 
- [x] delete document

### CODE REVIEW

**Comment by mfts** [comment]
Thanks for the PR @Aashish-Upadhyay-101  will review and merge this one tomorrow :)

File: pages/api/documents/[id]/index.ts [hunk 1]
```diff
@@ -40,9 +41,53 @@ export default async function handle(
...
+  
+      // delete the document from vercel blob 
+      await del(document.file);
+
+      // delete the document from database
+      await prisma.document.delete({
+        where: {
+          id: id,
+        }
+      })
```
Comment (mfts) [inline pages/api/documents/[id]/index.ts]: I would suggest wrapping both of these in a Promise.all

File: pages/api/documents/[id]/index.ts [hunk 2]
```diff
@@ -40,9 +41,53 @@ export default async function handle(
...
+      return res.status(500).json({
+        message: "Internal Server Error",
+        error: (error as Error).message,
+      });
+    }
+
   } else {
-    // We only allow GET and POST requests
-    res.setHeader("Allow", ["GET"]);
+    // We only allow GET, DELETE and POST requests
```
Comment (mfts) [inline pages/api/documents/[id]/index.ts]:
```suggestion
    // We only allow GET and DELETE requests
```

I realized that there's no POST request in this API route

**Review by mfts — CHANGES_REQUESTED** [review]

**Comment by Aashish-Upadhyay-101** [comment]
@mfts new changes : )

**Review by mfts — APPROVED** [review]
great work @Aashish-Upadhyay-101 !

**Pull request merged**